framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,324 @@
+     *     value of the two ranges
+     */
+    public Range union(Range right) {
+        long resultFrom = Math.min(from, right.from);
+        long resultTo = Math.max(to, right.to);
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Intersects two ranges. If there is no overlap between two ranges, a abnormal range with from
+     * greater than to would be returned. This would be caught by
+     * ValueAnnotatedTypeFactory.createIntRangeAnnotation when creating an annotation from range,
+     * which would then be replaced with @UnknownVal
+     *
+     * @param right the range to intersect with this range
+     * @return a range from
+     */
+    public Range intersect(Range right) {
+        long resultFrom = Math.max(from, right.from);
+        long resultTo = Math.min(to, right.to);
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Adds one range to another.
+     *
+     * @param right the range to be added to this range
+     * @return the smallest single range that includes all possible values resulted from adding any
+     *     two values selected from two ranges respectively
+     */
+    public Range plus(Range right) {
+        long resultFrom = from + right.from;
+        long resultTo = to + right.to;
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Subtracts one range from another
+     *
+     * @param right the range to be subtracted from this range
+     * @return the smallest single range that includes all possible values resulted from subtracting
+     *     an arbitrary value in @param from an arbitrary value in this range
+     */
+    public Range minus(Range right) {
+        long resultFrom = from - right.to;
+        long resultTo = to - right.from;
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Multiplies one range by another
+     *
+     * @param right the range to be multiplied by this range
+     * @return the smallest single range that includes all possible values resulted from multiply an
+     *     arbitrary value in @param by an arbitrary value in this range
+     */
+    public Range times(Range right) {
+        long[] possibleValues = new long[4];
+        possibleValues[0] = from * right.from;
+        possibleValues[1] = from * right.to;
+        possibleValues[2] = to * right.from;
+        possibleValues[3] = to * right.to;
+        return getRangeFromPossibleValues(possibleValues);
+    }
+
+    /**
+     * Divides one range by another
+     *
+     * @param right the range to divide this range
+     * @return the smallest range that includes all possible values resulted from dividing an
+     *     arbitrary value in this range by an arbitrary value in @param
+     */
+    public Range divide(Range right) {
+        long resultFrom = Long.MIN_VALUE;
+        long resultTo = Long.MAX_VALUE;
+
+        // TODO: be careful of divided by zero!
```

Comment:
Can you document what this does if divide by zero is possible given the range?  Or do you just mean that you want to avoid a divde-by-zero exception in this code?